docs/bricks_design.rst

```diff
@@ -0,0 +1,602 @@
+
+Bricks design explained
+=======================
+
+This is a runnable tutorial, which means you can copy its fragments 
+to an IPython notebook and run. The import section follows:
+
+    >>> import theano
+    >>> theano.config.compute_test_value = 'raise'
+    >>> from theano import tensor
+
+Please keep in mind that the code in this tutorial was written for illustration
+purposes only. Often only the common case is handled and corner cases are
+ignored.
+
+Introduction
+------------
+
+Deep learning (DL) models implementing using Theano share numerous
+building blocks which become more and more standartized over time.
+Factoring out such components is a very lucrative idea, and many people
+have fallen victims to the temptation to develop a library of DL
+building blocks. The list of rather well-known projects includes
+`PyLearn2`_, `Groundhog`_, `Lasagne`_, `Keras`_ and of course Blocks. Many people and
+research groups have smaller and more specialized frameworks.
+
```

Comment:
Still, I feel that we may need to showcase more the design assumptions. If you look into Lasagne, Keras, https://github.com/Ivaylo-Popov/Theano-Lights they all make a point of being simple. Blocks are more complicated. We need to justify it - Blocks are complicated, because it allows you to do X, Y, and Z. Because they allow you to do complicated stuff.

However, you are right that this may not be the best place for it. My comment was aimed at someone with some programming experience who wants to select a library for his project. You aim to explain Blocks to someone who has chosen them, but doesn't understand them (like a new lab member).

So maybe it is best to say to whom you address this tutorial?